devito/finite_differences/tools.py

```diff
@@ -139,6 +140,83 @@ def diff_f(expr, deriv_order, dims, fd_order, side=None, **kwargs):
     return derivatives
 
 
+class IndexSet(tuple):
+
+    """
+    The points of a finite-difference expansion.
+    """
+
+    def __new__(cls, dim, indices=None, expr=None, fd=None):
+        if fd is None:
+            try:
+                v = {d for d in expr.free_symbols if isinstance(d, StencilDimension)}
```

Comment:
So should we just check first `(fd or expr) is not None`

Reply:
added